src/manager.js

```diff
@@ -1,454 +1,420 @@
-        slotsToLoad.push(slot);
-      }
-      googletag.display(element.id);
-    }
-    if(!this.debug) {
-      googletag.pubads().refresh(slotsToLoad);
+    if ((thisEl.getAttribute('data-ad-load-state') === 'loaded') || (thisEl.getAttribute('data-ad-load-state') === 'loading')) {
+      continue;
     }
 
-    // We need to call the slot render ended stuff ourselves in debug mode.
-    if (this.debug) {
-      for(i = 0; i < slotsToLoad.length; i++) {
-        slot = slotsToLoad[i];
-        var domId = slot.getSlotId().getDomId();
-        var iframe = document.getElementById(domId).getElementsByTagName('iframe')[0];
-        if (!iframe) {
-          return;
-        }
-        var fakeEvent = {
-          'slot': slot,
-          'size': [parseInt(iframe.width, 10), parseInt(iframe.height, 10)]
-        };
-        this.onSlotRenderEnded(fakeEvent);
-      }
-    }
-  },
+    var slot = this.configureAd(thisEl);
 
-  unloadAds: function (el) {
-    if (! this.initialized) {
-      return;
+    if (slot) {
+      slotsToLoad.push(slot);
     }
-    // Unload the ads, likely for performance reasons.
-    var ads = this.findAds(el);
-    var ad, i;
-    if (!this.debug) {
-
-      // If we're not in debug mode, get a list of the slot objects
-      var slots = [];
-      for (i = 0; i < ads.length; i++) {
-        ad = ads[i];
-        // Lock the height...
-        ad.style.height = ad.offsetHeight + 'px';
-        ad.style.width = ad.offsetWidth + 'px';
-
-        if (ad.id in this.slots) {
-          slots.push(this.slots[ad.id]);
-          delete this.slots[ad.id];
-        }
-
-        ad.setAttribute('data-ad-load-state', 'unloaded');
-      }
-      googletag.pubads().clear(slots);
+    this.googletag.display(thisEl.id);
+    thisEl.setAttribute('data-ad-load-state', 'loading');
+  }
 
-    } else {
-      for (i = 0; i < ads.length; i++) {
-        ad = ads[i];
-        if (ad.id in this.slots) {
-          while (ad.hasChildNodes()) {
-            ad.removeChild(ad.firstChild);
-          }
-          delete this.slots[ad.id];
-        }
-        ad.setAttribute('data-ad-load-state', 'unloaded');
-      }
+
+  if (slotsToLoad.length > 0) {
+    this.googletag.pubads().refresh(slotsToLoad);
+  }
+};
+
+/**
+ * Unloads all ads
+ *
+ * @param {Element} optional element to scope where to unload ads in the document
+ * @returns undefined
+*/
+AdManager.prototype.unloadAds = function(element) {
+  if (! this.initialized) {
+    return;
+  }
+
+  var ads = this.findAds(element);
+  var ad, i;
+
+  var slots = [];
+  for (i = 0; i < ads.length; i++) {
+    ad = ads[i];
+    ad.style.height = ad.offsetHeight + 'px';
+    ad.style.width = ad.offsetWidth + 'px';
+
+    if (ad.id in this.slots) {
+      slots.push(this.slots[ad.id]);
+      delete this.slots[ad.id];
     }
+
+    ad.setAttribute('data-ad-load-state', 'unloaded');
+  }
+  this.googletag.pubads().clear(slots);
+};
+
+var AdManagerWrapper = {
+  init: function(options) {
```

Comment:
I could refactor this to use the singleton pattern and then this would work as intended